src/lib/comms/include/sol-http-client.h

```diff
@@ -92,7 +92,7 @@ struct sol_http_request_interface {
      *       inside this callback.
      */
     ssize_t (*on_data)(void *userdata, struct sol_http_client_connection *connection,
-        const struct sol_buffer *buffer);
+        struct sol_buffer *buffer);
```

Comment:
Taking a better look I think it's better let it const. My points are:

- The memory inside buffer is `const` (allocated by curl), so the user will not be able to change it.
- Without the `const` is not clear who is responsible to call `sol_buffer_fini`. Now the caller is calling the `sol_buffer_fini` after the callback.
- Functions to get/see the buffer contents already receive a `const struct sol_buffer`. i.e `sol_buffer_at`, `sol_buffer_get_slice` ...

If you still think it's better remove the const, it will be necessary change ./src/samples/http/download.c too.